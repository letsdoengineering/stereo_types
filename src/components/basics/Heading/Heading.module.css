.title {
    font-family: "Lucida Sans", "Lucida Grande", Arial, sans-serif;
    color: var(--dark-gray);
}

.title-1 {
    font-family: "Lucida Sans", "Lucida Grande", Arial, sans-serif;
    color: var(--default-blue);
    font-size: 2rem;
}

.title-2 {
    font-size: 1.25rem;
}

.title-3 {
    font-size: 1.125rem;
}

.title-4 {
    font-size: 1rem;
}

.title-5 {
    font-size: 1rem;
}

.title-6 {
    font-size: 1rem;
}

.weight-normal {
    font-weight: normal;
}

.weight-bold {
    font-weight: bold;
}

.weight-xtra {
    font-weight: 900;
}
