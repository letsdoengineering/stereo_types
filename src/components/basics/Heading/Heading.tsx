import React from 'react'
import getClassNames from '../../../utils/get-class-names'
import * as headingStyles from './Heading.module.css'

// TODO: Find appropriate interface to extend for h1-h6
type HeadingProps = {
  /** the section heading tag level between <h1>-<h6> that you wish to set */
  level?: '1' | '2' | '3' | '4' | '5' | '6'
  /** optional: class-names */
  className?: string
  /** font weight */
  weight?: 'Normal' | 'Bold' | 'Xtra'
  /** children always of type ReactNode */
  children: React.ReactNode
}

/** Heading: component that returns a section heading tag between <h1>-<h6>
 * defaults to H1 if nothing is set */
const Heading: React.FC<HeadingProps> = ({
  level = '1',
  className,
  weight = 'Normal',
  children,
  ...rest
}: HeadingProps) => {
  const classNames = getClassNames({
    defaultClasses: [
      headingStyles.title,
      headingStyles[`title${level}`],
      headingStyles[`weight${weight}`],
    ],
    className,
  })
  const Wrapper = `h${level}` as keyof JSX.IntrinsicElements

  return (
    <Wrapper className={classNames} {...rest}>
      {children}
    </Wrapper>
  )
}

export default Heading
