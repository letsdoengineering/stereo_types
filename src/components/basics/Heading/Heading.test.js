import * as React from 'react'
import { render, screen, waitFor } from '@testing-library/react'
import Heading from './Heading'
import renderer from 'react-test-renderer'
import { axe } from 'jest-axe'

describe('Heading', () => {
  describe('dynamic header levels', () => {
    it('renders a h1 element by default', async () => {
      render(<Heading>Some header</Heading>)
      const header = screen.getByText('Some header')

      await waitFor(() => {
        expect(header).toBeInTheDocument()
        expect(header.nodeName).toBe('H1')
      })
    })

    it('renders a h3 element if level prop is specified as 3', async () => {
      render(<Heading level='3'>Some header</Heading>)
      const header = screen.getByText('Some header')

      await waitFor(() => {
        expect(header).toBeInTheDocument()
        expect(header.nodeName).toBe('H3')
      })
    })
  })

  it('has not changed the JSX layout of the page from previous snapshot', () => {
    const tree = renderer.create(<Heading>Some header</Heading>).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders the page DOM that is accessible', async () => {
    const { container } = render(<Heading>Some header</Heading>)
    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
