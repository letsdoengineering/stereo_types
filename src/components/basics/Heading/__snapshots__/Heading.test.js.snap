// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading has not changed the JSX layout of the page from previous snapshot 1`] = `
<h1
  className="title title1 weightNormal"
>
  Some header
</h1>
`;
